bindings/rascal/models/krr.py

```diff
@@ -74,11 +75,16 @@ def predict(self, managers, compute_gradients=False):
         np.array
             predictions
         """
-        KNM, Y0 = self._preprocess_input(managers, compute_gradients)
+
         if compute_gradients is False:
+            KNM, Y0 = self._preprocess_input(managers, compute_gradients)
```

Comment:
Now that the kernel gradient predictions are moved into a separate function, wouldn't it be a good idea to do the same with forces? I'm a bit confused as to why this function returns _either_ energies _or_ forces, and not both (when you want the forces, you usually also want the energies, right?). It seems a better idea to split this up into two functions, one that computes energies and one that computes forces (gradients) rather than switching between two essentially separate functions in here.

Reply:
I don't see the difference between this way and having separate routines. I would rather not make such change now as it would require many modifications of the examples and tests that are quite unrelated to the PR.

Reply:
The difference is: Two separate, well-documented functions, each doing one thing, returning a single array of predictable shape, versus this, which is just two functions rolled into one and returns a different thing depending on a boolean switch in the input. I know there are functions especially in the NumPy library that do this sort of thing (return different things depending on input switches), but (1) they usually only do this for returning _extra_ information besides the main return array, and (2) it's unpredictable and confusing for new users.

As for the separate examples and tests: They're not "unrelated" to this PR if they have to do with property (and gradient) prediction. And you can always make this into a "compatibility" function that just delegates to two functions that do the work, then mark this as deprecated.

Reply:
this issue will be dealt with in #300.